### PR TITLE
TYP: misc property return types

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

File: pandas/core/arrays/_mixins.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def _validate_fill_value(self, fill_value):
     # TODO: make this a cache_readonly; for that to work we need to remove
     #  the _index_data kludge in libreduction
     @property
-    def shape(self) -> Tuple[int, ...]:
+    def shape(self) -> Tuple[int]:
```
Comment (jbrockmendel) [inline pandas/core/arrays/_mixins.py:71]: datetimearray, timedeltaarray can be 2d sometimes

Comment (simonjayhawkins) [reply pandas/core/arrays/_mixins.py:71]: is that accounted for in the changes further down to pandas/core/internals/blocks.py

Comment (jbrockmendel) [reply pandas/core/arrays/_mixins.py:71]: blocks shouldnt come in to it.  Part of the idea of the mixin in this file is to support 2D EAs.

Comment (simonjayhawkins) [reply pandas/core/arrays/_mixins.py:71]: the backing data could be 2d? but isn't the EA only ever 1d?

Comment (jbrockmendel) [reply pandas/core/arrays/_mixins.py:71]: There are corner cases where a DatetimeArray/TimedeltaArray can be 2D in ops.  (and if i get my way, eventually we'll support 2D in general)

Comment (simonjayhawkins) [reply pandas/core/arrays/_mixins.py:71]: ok i'll revert some of these changes for now. the base array class will need to allow a 2 tuple then. rather than Tuple[int, ...] we may want to go with Union[Tuple[int] ,Tuple[int, int] ] to only allow 1d and 2d

Comment (jreback) [reply pandas/core/arrays/_mixins.py:71]: might be worth having an alias: Shape1D, Shape2D or similar

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -3123,13 +3122,13 @@ def shape(self):
...
+                shape_list = list(shape[0 : (ndim - 1)])
             else:
-                shape = []
+                shape_list = []
 
-            shape.append(items)
+            shape_list.append(items)
 
-            return shape
+            return tuple(shape)
```
Comment (simonjayhawkins) [inline pandas/io/pytables.py:3131]: I made a mistake here and tests didn't fail. so will revert the changes to pandas/io/pytables.py for now.

File: pandas/core/arrays/_mixins.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/_mixins.py:71]: ended up going down a rabbit hole on this. so will come back another day refreshed.

**Comment by jbrockmendel** [comment]
LGTM

**Review by WillAyd — APPROVED** [review]
lgtm as well - I think can merge once you take out of draft

**Comment by simonjayhawkins** [comment]
closing to clear queue
